### ISSUE TITLE
Return the appropriate username during mail autoconfig

### ISSUE THREAD

**Issue by zamentur** [YunoHost/issues#1707]
We still have an autoconfig mechanism for email client configuration.

However if the local part of the email is not equal to the username, the autoconfiguration fails.

Exemple: 
```
username : camille
email: camille.doe@example.com
```

The autoconfig requested is 
```
GET /.well-known/autoconfig/mail/config-v1.1.xml?emailaddress=camille.doe%40example.com
```

But our current config return in username tags:
```
<username>%EMAILLOCALPART%</username>
```
=> camille.doe
=> So the autoconfig fails

If the autoconfig was generated dynamically with PHP or another techno (lua ?), it could be:
```
<username>camille</username>
```

Doc: https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration/FileFormat/HowTo#Username

**Comment by alexAubin** [comment]
(Alternatively we could decide to switch to login = the full email address, dunno if that's better or worse)

**Comment by npalix** [comment]
In that case, it should exist a 'camille@example.com' canonical address used for the email client configuration.
'camille.doe@example.com' will be a user defined alias if requested by the user. 

From a RFC point of view,
https://datatracker.ietf.org/doc/html/draft-daboo-srv-email-02#section-5
"IMAP, POP3, and SMTP (submission) servers SHOULD be configured to
       allow authentication with email addresses or email local-parts."

According to,
https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat
The goal is to stick to a static file, and the 'right' value for that use case seems to be either %REALNAME% or %USERNAME%. 

So, pick your solution, but their is not need to make it dynamic.

**Comment by zamentur** [comment]
> In that case, it should exist a '[camille@example.com](mailto:camille@example.com)' canonical address used for the email client configuration.
> '[camille.doe@example.com](mailto:camille.doe@example.com)' will be a user defined alias if requested by the user.
It's the case when the user is created. But the admin can change the email with `yunohost user update`.

So you suggest we remove the possibility to change the first email of a user with `yunohost user update` ?

**Comment by npalix** [comment]
It sound good for me to preserve the canonical address, and it's a simple way to ensure the autoconfig is working
without changing it.

If the main email address is editable, then an alternative is to present the connection information to pass to the user (full email address) as the email server "SHOULD" allow email-based authentication. But the email server configuration must be checked then (is it currently working?). Finally, the autoconfig can switch to "%EMAILADDRESS%" instead of "%EMAILLOCALPART%. That way the autoconfig remains static.

**Comment by eldertek** [comment]
I think should be closed as autoconfig in this case working.

**Comment by zamentur** [comment]
I agree with you in 12.1 it's possible to login with any alias of the ldap account.
```
            user = _get_ldap_interface().search("ou=users", f"mail={username}", ["uid"])
```
